aspnet/publishing/iis.rst

```diff
@@ -1,116 +1,187 @@
 Publishing to IIS
-=============================
+=================
 
-By `Rick Anderson`_ and `Luke Latham`_
+By `Rick Anderson`_ and `Luke Latham <https://github.com/GuardRex>`_
 
-.. contents:: Sections:
-  :local:
-  :depth: 1
+- `Supported Operating Systems`_
+- `IIS Configuration`_
+- `Install the Microsoft ASP.NET Core Module`_
+- `Install .NET Core on the Server`_
+- `Deploy the Application`_
+- `Configure the Website in IIS`_
+- `Create a Data Protection Registry Hive`_
+- `Common Errors`_
+- `Additional Resources`_
 
-Install the HTTP Platform Handler
------------------------------------
+Supported Operating Systems
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-- Install the HTTP Platform Handler version 1.2 or higher:
+The following operating systems are supported:
 
-  - `64 bit HTTP Platform Handler <http://go.microsoft.com/fwlink/?LinkID=690721>`_
-  - `32 bit HTTP Platform Handler <http://go.microsoft.com/fwlink/?LinkId=690722>`_
+- Windows 7 and newer
+- Windows Server 2008 R2 and newer\*
 
-If you need to enable IIS, see `IIS server configuration`_.
+\*Conceptually, the IIS configuration described in this document also applies to hosting ASP.NET Core applications on Nano Server IIS, but refer to `ASP.NET Core on Nano Server <https://docs.asp.net/en/latest/tutorials/nano-server.html>`_ for specific instructions.
 
-Configure Data Protection
------------------------------------
+IIS Configuration
```

Comment:
[What follows is a little off-topic ... and a **BOOK**, but humor me for a minute. I just want to get my thoughts down and out on this. As an educator, this is important to me. Read the TL;DR and skip it. I just want to say my peace and rest in peace :skull:. I know a dude ... a dude who was big at MS back in the '90's ... he is good friends with Satya ... yeah ... "a dude." He just told me the other day that MS has the same challenges that all giant corporations have. Things like politics, difficulty in communication, difficulty in executing large-scale strategies, and difficulty holding people accountable are all alive and well there to spite best efforts. It's easy for me to sit out here and shoot my mouth off. I get it. However, you did OSS this. So ... read the TL;DR and skip the rest. I'm going on the record though with **one** of my pet peeves about MS docs across the company.]

TL;DR ... I'll chop the IIS Config here and chop the purely IIS Troubleshooting bits; however, please consider a full tutorial `ASP.NET Core on IIS` that keeps **ALL** of this stuff in one place and adds a little more **cowbell** :bell:.

If not, ok then ... @benaadams just suggested to me ... blog it and those of us on the "JabbR Outreach User Support Team" (JOUST in superhero graphic novel circles) can point to the GuardRex blog. We can do that. That sucks ... not what I want to do ... but MS is MS and GuardRex is GuardRex, and I know what I can control. :smile:

> we don't want to maintain docs for technologies that we don't directly own.

... which I take as a flaw in the organization of Microsoft business units, philosophy, and effort to find a solution. I doubt that this is an intractable problem.

I have a very strong suspicion that if we divided up a large group of devs randomly into two groups and gave one group this doc in its present form and gave the other group this doc with the IIS bits (IIS config and purely IIS troubleshooting cut out), that the first group would :guitar: **ROCK** :guitar:. <---- That should be the guide ... what works for the **customer** ... not what is expedient or simple for the company. I suspect MS will lose far more money in operational support with crappy docs than it will by investing in a solution to this problem.

Sure, IIS can throw many more exceptions than what I show here. I was just hitting the common ones that I've seen devs messing up in the trenches over the past year. If there is a great IIS doc that runs down all of the basic IIS screw-ups a dev can make and what to do about them, that doc should be linked here with no mention of those problems here. One little problem though ... that doc **doesn't exist** AFAIK. You want to link this??? ... http://www.iis.net/learn/troubleshoot ??? Yuck. Which IIS doc covers the minimal IIS config for .NET Core app hosting on WS IIS and local IIS? Which one? ... Yeah ... It doesn't exist. It's only here in this revision.

It's inefficient to create unnatural breaks in docs because one team "owns" a piece of a process and some other team "owns" another piece. There has to be an approach/philosophy/process/engineering solution that would allow the IIS folks to "own" and be responsible for their pieces of this doc (the purely config and troubleshooting bits) while leaving all of this content accessible here, which is a one-stop-shop for ASP.NET Core on IIS with the most common dev screw-ups based on working with devs in the field.

If this doc is cut without a comprehensive tutorial, then devs will probably suffer. MS will probably lose money. .NET Core on IIS will be a sour first experience for many. I'm telling you from the trenches on JabbR that I have been in conversations with frustrated devs, and I've seen them walking away from things when the docs (because we're in RC) don't support what they are trying to implement.

I've seen @davidfowl spend hours re-explaining things, including IIS config. One of his recent conversations re-explaining something was 1.5 hours. I timed it. We were laughing about it afterwards to reduce the pain. The current version of the IIS revision is getting hits right now. It's actually (as @Aequoris just said) useful in its current comprehensive form. He's not the only one: Two more devs on JabbR said they fixed their IIS hosting probs based on this doc revision. This has nothing to do with .NET Core in RC2 [or RC17 (haha)] ... if this doc in its present form saves @davidfowl even 10 minutes, it was worth it leaving the IIS bits in it.

The problem here is the same old problem truly going back to 1975! MS engineers are great at building things and they often suck at explaining to end-users how to use things. Documentation is still far-too-frequently an after-thought by engineers and teams. Oftentimes, the team just doesn't have the communications "chops," willpower, or available time to do the job properly.

Take the Nano Server Team for example: They are **insanely GREAT engineers** ... I :heart: those :cat:s and Nano is **amazing technology**, but their docs are insanely **bad**. Why? Plain and simple, they're just not that good at explaining things to people. I guess that like many high-end engineers that they went into technology to build things and not to teach things. The lack of experience in education and communication combined with little desire to sit for many hours working on a single doc is obvious. I spent four full days on this revision. Can you really see any of them with their interest in engineering and their backlog being able to do that? No way. That's not really a knock on them. I think this is a fact of business unit organization there very heavily ... too heavily as far as docs are concerned ... centered on engineering. I think the problem is actually a lack of creative thinking on how to support documentation in a giant organization.

Maybe Apple has figured it out ... maybe Adobe has figured it out. idk ... but Microsoft surely hasn't figured it out, or else we wouldn't be talking about cutting **critical** parts of this doc without addressing the developer pain that will ensue. We wouldn't see MS using at least a dozen completely different doc platforms across the company that have no other connection than web links (and they're poorly linked at that). [Ok ... ok ... I'm being a little melodramatic here! :smile: ... one gets so few chances to wax poetic in tech circles these days. lol]

Put the same creative juice that goes into MS products and services into docs. What is needed is some lateral thinking here. The model of each technology unit "owning" its own docs and never talking about another unit's technology is clearly not working for many types of cross-technology-cutting docs, introductory docs, tutorials, and samples.

... and they lived happily ever after. THE END